### PR TITLE
[SPARK-50283][INFRA] Add a separate docker file for linter

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Add a separate docker file for linter

### Why are the changes needed?
1, to centralize the installation of linter;
2, to spin it off the single docker;

### Does this PR introduce _any_ user-facing change?
no, infra-only

### How was this patch tested?
ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: python/pyspark/sql/connect/expressions.py [hunk 1]
```diff
@@ -490,7 +490,7 @@ def __repr__(self) -> str:
                 # is sightly different:
                 # java.time.Duration only applies HOURS, MINUTES, SECONDS units,
                 # while Pandas applies all supported units.
-                return pd.Timedelta(delta).isoformat()  # type: ignore[attr-defined]
```
Comment (zhengruifeng) [inline python/pyspark/sql/connect/expressions.py:493]: it is surprising that the two `type: ignore` were caused by non-linter dependency.

**Comment by zhengruifeng** [comment]
also cc @panbingkun

File: python/pyspark/sql/connect/expressions.py (hunk 1 above)
Comment (dongjoon-hyun) [reply python/pyspark/sql/connect/expressions.py:493]: Oh, good to know. I also didn't expect those kind of situations.

File: .github/workflows/build_infra_images_cache.yml [hunk 2]
```diff
@@ -74,3 +75,16 @@ jobs:
       - name: Image digest (Documentation)
         if: hashFiles('dev/spark-test-image/docs/Dockerfile') != ''
         run: echo ${{ steps.docker_build_docs.outputs.digest }}
+      - name: Build and push (Linter)
+        if: hashFiles('dev/spark-test-image/lint/Dockerfile') != ''
+        id: docker_build_lint
+        uses: docker/build-push-action@v6
+        with:
+          context: ./dev/spark-test-image/docs/
```
Comment (LuciferYang) [inline .github/workflows/build_infra_images_cache.yml:83]: Should it be  `./dev/spark-test-image/lint/` ？

**Review by HyukjinKwon — APPROVED** [review]

File: .github/workflows/build_infra_images_cache.yml (hunk 2 above)
Comment (zhengruifeng) [reply .github/workflows/build_infra_images_cache.yml:83]: good cache

**Review by LuciferYang — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**